### PR TITLE
Fix ibmq_connector tests for premium

### PR DESCRIPTION
**Pull request by diego-plan9**
Adjust the ibmq_connector tests for passing in premium:
* use the right backend depending on the kind of credentials in a
  number of tests
* disable `available_backend_simulators` test, as the function is not
  used in terra and seems not ready for premium.

This is a fix for the tests - further cleanup will be handled in #1175.

### CODE REVIEW

**Comment by jaygambetta** [comment]
I dont even know what the `available_backend_simulator` is

**Comment by diego-plan9** [comment]
> I dont even know what the available_backend_simulator is

Yes, it seems it was a part of `IBMQuantumExperience` that was used some time ago but not anymore (at least in terra) - I'll tackle it in the next round.

**Comment by jaygambetta** [comment]
i think we should just remove it all together.

File: test/python/ibmq/test_ibmq_connector.py [hunk 1]
```diff
@@ -234,7 +245,8 @@ class TestAuthentication(QiskitTestCase):
     @requires_qe_access
     def test_url_404(self, qe_token, qe_url):
         """Test accessing a 404 URL"""
-        url_404 = '{}/API_TEST'.format(qe_url)
+        url_404 = re.sub(r'\/api.*$', '/api/TEST_404', qe_url)
+        print(url_404)
```
Comment (delapuente) [inline test/python/ibmq/test_ibmq_connector.py]: I think this slipped into the PR unintentionally.

**Comment by jaygambetta** [comment]
i  removed the print @delapuente could you look now. I cant approve now :-(

**Pull request merged**